### PR TITLE
Added CheckHTTP in flowcontrol with Gateway integration support

### PR DESCRIPTION
**Pull request by hdkshingala**
image

image

### Summary by OpenAI

New Feature: Added CheckHTTP service to perform Aperture Flowcontrol check on incoming requests to a networked service. Includes support for Gateway integration with Kong API Gateway and Nginx plugin, as well as new HTTP request and response messages and enums. Also adds CircleCI job for publishing an Aperture Lua module and updates to documentation.

### CODE REVIEW

File: pkg/otelcollector/metricsprocessor/internal/envoy_labels.go
Comment (harjotgill) [inline pkg/otelcollector/metricsprocessor/internal/envoy_labels.go:1]: @openai can you please summarize changes made to this file?

File: pkg/policies/flowcontrol/service/checkhttp/checkhttp-labels.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	case "HTTP/1.1":
+		return "1.1"
+	case "HTTP/2":
+		return "2.0"
+	case "HTTP/3":
+		return "3.0"
+	default:
+		return protocolName
+	}
+}
```
Comment (harjotgill) [inline pkg/policies/flowcontrol/service/checkhttp/checkhttp-labels.go:85]: @openai,
Describe the changes in this file

Comment (harjotgill) [reply pkg/policies/flowcontrol/service/checkhttp/checkhttp-labels.go:85]: What are the exported functions on this file?
Can you provide the comments threat I can add on top of each exported function? Please provide a bullet point list of functions and comments that I can add.

File: pkg/policies/flowcontrol/servicegetter/servicegetter.go [hunk 2]
```diff
@@ -132,3 +134,8 @@ func (sg emptyServiceGetter) ServicesFromContext(ctx context.Context) []string {
 func (sg emptyServiceGetter) ServicesFromSocketAddress(addr *corev3.SocketAddress) []string {
 	return nil
 }
+
+// ServicesFromAddress implements ServiceGetter interface for emptyServiceGetter.
+func (sg emptyServiceGetter) ServicesFromAddress(addr string) []string {
+	return nil
+}
```
Comment (harjotgill) [inline pkg/policies/flowcontrol/servicegetter/servicegetter.go:141]: @openai can you please implement `ServiceFromAddress` function?

File: pkg/policies/flowcontrol/servicegetter/servicegetter.go
Comment (harjotgill) [inline pkg/policies/flowcontrol/servicegetter/servicegetter.go:1]: @openai please draft a unit test plan to test the code in this file

Comment (harjotgill) [reply pkg/policies/flowcontrol/servicegetter/servicegetter.go:1]: Based on this test plan, please generate a unit test plan file in golang.

Comment (hdkshingala) [reply pkg/policies/flowcontrol/servicegetter/servicegetter.go:1]: Added there tests with minor changes required for configuring entities.

File: pkg/otelcollector/metricsprocessor/internal/lua_labels.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package internal
```
Comment (hdkshingala) [inline pkg/otelcollector/metricsprocessor/internal/lua_labels.go:1]: @openai please draft a unit test plan to test the code in this file

Comment (hdkshingala) [reply pkg/otelcollector/metricsprocessor/internal/lua_labels.go:1]: Based on this test plan, please generate a unit test plan file in golang.

File: .circleci/post-release.yaml [hunk 4]
```diff
@@ -266,6 +266,23 @@ jobs:
...
+      - checkout
+      - run: *install-gh-cli
+      - run:
+          name: Attach package to release
+          command: |
+            tar -czvf aperture-lua.tar.gz gateways/lua/
+            gh release upload "${VERSION}" "aperture-lua.tar.gz" --repo=fluxninja/aperture
+
 workflows:
   version: 2
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply .circleci/post-release.yaml:288]: It is covered as part of documentation

**Pull request merged**